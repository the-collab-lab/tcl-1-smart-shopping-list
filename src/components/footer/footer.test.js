import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import Footer from './index';
import { SmartLink } from '../index';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <Footer />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot on load', () => {
  const tree = TestRenderer.create(
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  );
  expect(tree.toJSON()).toMatchSnapshot();
});

it('confirm the two expected links are present', () => {
  const tree = TestRenderer.create(
    <BrowserRouter>
      <Footer />
    </BrowserRouter>
  );

  const footerChildren = tree.toJSON().children;
  expect(footerChildren.length).toBe(2);
  expect(footerChildren[0].children[0]).toBe('My List');
  expect(footerChildren[1].children[0]).toBe('Add Item');

  expect(tree.toJSON()).toMatchSnapshot();
});
