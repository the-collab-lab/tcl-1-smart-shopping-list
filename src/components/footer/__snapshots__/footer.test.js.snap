// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot on load 1`] = `
<footer
  className="footer"
>
  <a
    href="/add-item"
    onClick={[Function]}
  >
    <button
      type="submit"
    >
      Add Item
    </button>
  </a>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      type="submit"
    >
      My List
    </button>
  </a>
</footer>
`;
